### ISSUE TITLE
Use libgcc.a compiled with -Os instead of the default -O2 on boards that are tight on space

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#4705]
The official ARM gcc toolchain compiles `libgcc.a` as `-O2`. The `arch` Linux ARM toolchain uses `-Os`. We discovered that building on `arch` saved a kB or two (at least).

We can consider replacing the stock `libgcc.a` in the ARM toolchain with a `-Os` version to make room on the smallest boards for new features.

**Comment by tannewt** [comment]
@dhalbert You look into this a bit right? What did you run into?

**Comment by dhalbert** [comment]
I tried to build `libgcc.a` as part of building the ARM gcc toolchain from their supplied sources, so I could then change `libgcc`s build flags and make sure I got a library that would be cleanly substitutable. Despite following the directions carefully, I was unable to get the toolchain to build far enough to build that library. I think I tried this twice. Perhaps it will be easier with the 11.2 sources
